sssd_test_framework/hosts/client.py

```diff
@@ -0,0 +1,135 @@
+"""IPA multihost host."""
+
+from __future__ import annotations
+
+from pytest_mh.ssh import SSHLog
+
+from .base import BaseBackupHost
+
+__all__ = [
+    "ClientHost",
+]
+
+
+class ClientHost(BaseBackupHost):
+    """
+    SSSD client host object.
+
+    Provides features specific to SSSD client.
+
+    .. note::
+
+        Full backup and restore of SSSD state is supported.
+    """
+
+    def pytest_setup(self) -> None:
+        """
+        Called once before execution of any tests.
+
+        - override systemd unit to disable burst limiting, otherwise we will be
+          unable to restart the service frequently
+        - reload systemd to apply change to the unit file
+        """
+        super().pytest_setup()
+
+        self.logger.info("Creating SSSD systemd override")
+        self.ssh.run(
+            """
+            set -ex
+
+            backuppath="/tmp/mh.client.systemd.backup"
+            overridedir="/etc/systemd/system/sssd.service.d"
+            if [ -d "$overridedir" ] || [ -f "$overridedir" ]; then
+                cp --force --archive "$overridedir" "$backuppath"
+            fi
+
+            # Disable burst limiting to allow often sssd restarts for tests
+            mkdir -p "$overridedir"
+            cat <<EOF > "$overridedir/override.conf"
+            [Unit]
+            StartLimitIntervalSec=0
+            StartLimitBurst=0
+            EOF
+
+            # Reload systemd to pickup changes
+            systemctl daemon-reload
+            """,
+            log_level=SSHLog.Error,
+        )
+
+    def pytest_teardown(self) -> None:
+        """
+        Called once after all tests are finished.
+
+        - remove systemd changes
+        """
+        self.logger.info("Removing SSSD systemd override")
+        self.ssh.run(
+            """
+            set -ex
+
+            backuppath="/tmp/mh.client.systemd.backup"
+            overridedir="/etc/systemd/system/sssd.service.d"
+
+            rm -fr "$overridedir"
+            if [ -d "$overridedir" ] || [ -f "$overridedir" ]; then
+                mv --force "$backuppath" "$overridedir"
+            fi
```

Comment:
Should we reload systemd to apply the changes?

Reply:
Yes, nice catch.